### PR TITLE
protect against corrupted Workflow::Models

### PR DESCRIPTION
**Pull request by nnutter**
If _find_first_nondone_step_impl crashes it can leave the related
Workflow::Models in a corrupt state that ends up having "invalid data for
save".  The (side-effect) changes are probably a bug and not relevant to
the task at hand so it's safest just to rollback if
_find_first_nondone_step_impl crashes.

The "invalid data for save" was already being rolled back later when the
transaction fails to commit but by pre-emptively doing it here we can save
a ton of time due to the slow performance of the "invalid data for save"
code.

### CODE REVIEW

**Comment by nnutter** [comment]
I'm also trying to see if I can improve the performance of "invalid data for save" and may also try to look into the side-effect changes but at the moment model-summary is taking three hours to run instead of three minutes so I'd like to get something out.

File: lib/perl/Genome/Model/Command/Admin/ModelSummary.pm [hunk 1]
```diff
@@ -324,12 +326,32 @@ sub status_compare { # http://stackoverflow.com/q/540229
...
 sub find_first_nondone_step {
+    my $build = shift;
+
+    # The following is wrapped in a transaction and try to protect against
+    # "corrupt" Workflow::Models.
+    my $tx = UR::Context::Transaction->begin();
+    my $first_nondone_step = try {
+        my $wf = $build->newest_workflow_instance;
+        my $step =  _find_first_nondone_step_impl($wf);
+        $tx->commit();
```
Comment (tmooney) [inline lib/perl/Genome/Model/Command/Admin/ModelSummary.pm]: Does this trigger the `ERROR: Invalid data for save!` message here instead of later?

Comment (nnutter) [inline lib/perl/Genome/Model/Command/Admin/ModelSummary.pm]: In our current problem `_find_first_nondone_step_impl()` is crashing so `$tx->commit()` does not run.

**Comment by nnutter** [comment]
I've been chasing rabbits... Tom's UR fix wasn't yet in master.   With the UR fix the performance of "invalid data for save" is amazingly better (1000x).  This PR may not be needed but I haven't decided whether it stand on its own merit or not.

**Comment by nnutter** [comment]
This PR does not significantly improve performance but does avoid the noise caused by Workflow::Models that can no longer correctly load due to removed (command) classes.

**Comment by tmooney** [comment]
+1 :palm_tree: Less noise is good!

**Comment by brummett** [comment]
+1

Alternatively, it could have used UR::Context::Transaction::do().  Though I see that eval_or_do() doesn't properly protect $@.  I'll fix that.

**Pull request merged**